auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -73,6 +74,13 @@
  * for examples on how to get started with FirebaseUI Auth.
  */
 public class AuthUI {
+    @StringDef({
+                       EmailAuthProvider.PROVIDER_ID, EMAIL_PROVIDER,
+                       GoogleAuthProvider.PROVIDER_ID, GOOGLE_PROVIDER,
+                       FacebookAuthProvider.PROVIDER_ID, FACEBOOK_PROVIDER,
+                       TwitterAuthProvider.PROVIDER_ID, TWITTER_PROVIDER
+               })
+    public @interface SupportedProvider {}
```

Comment:
Do we want this to be public? I feel like it could be useful, but it also means we have to maintain backwards compatibility...